lib/jsonld-signatures.js

```diff
@@ -19,31 +19,47 @@ const VerificationError = require('./VerificationError');
  * Cryptographically signs the provided document by adding a `proof` section,
  * based on the provided suite and proof purpose.
  *
- * @param {object|string} document - The document to be signed, either a string
- *  URL (resolved to an object via the given `documentLoader`) or a plain
- *  object (JSON-LD document).
+ * @param {object|string} document - The document to be signed, either
+ *   a string URL (resolved to an object via the given `documentLoader`) or a
+ *   plain object (JSON-LD document).
+ *
+ * @param {object} options - Options hashmap.
+ * @param {LinkedDataSignature} options.suite - The linked data signature
+ *   cryptographic suite, containing private key material, with which to sign
+ *   the document.
  *
- * @param {LinkedDataSignature} suite - The linked data signature cryptographic
- *   suite, containing private key material, with which to sign the document.
  * @param {ProofPurpose} purpose - A proof purpose instance that will
  *   match proofs to be verified and ensure they were created according to
  *   the appropriate purpose.
  *
+ * @param {function} documentLoader  - A secure document loader (it is
+ *   recommended to use one that provides static known documents, instead of
+ *   fetching from the web) for returning contexts, controller documents, keys,
+ *   and other relevant URLs needed for the proof.
+ *
  * Advanced optional parameters and overrides:
  *
- * @param {function} [documentLoader]  - A custom document loader,
- *   `Promise<RemoteDocument> documentLoader(url)`.
- * @param {function} [expansionMap] - A custom expansion map that is
+ * @param {function} [options.expansionMap] - A custom expansion map that is
  *   passed to the JSON-LD processor; by default a function that will throw
  *   an error when unmapped properties are detected in the input, use `false`
  *   to turn this off and allow unmapped properties to be dropped or use a
  *   custom function.
+ * @param {boolean} [options.addSuiteContext=true] - Toggles the default
+ *   behavior of each signature suite enforcing the presence of its own
+ *   `@context` (if it is not present, it's added to the context list).
  *
  * @returns {Promise<object>} Resolves with signed document.
  */
 api.sign = async function sign(document, {
-  suite, purpose, documentLoader, expansionMap
+  suite, purpose, documentLoader, expansionMap, addSuiteContext = true
 } = {}) {
+  // Ensure document contains the signature suite specific context url
+  // or throw an error (in case an advanced user overrides the `addSuiteContext`
+  // flag to false).
+  _ensureSuiteContext({
+    document, contextUrl: suite.contextUrl, addSuiteContext
+  });
```

Comment:
Yeah, this will need to be moved into `ProofSet.add` to handle the document as a string use case.